documentation/src/main/asciidoc/topics/query.adoc

```diff
@@ -18,480 +18,570 @@ Apart from indexed queries, {brandname} can run queries over non-indexed or
 
-NOTE: The presence of an `<indexing>` element which omits the `enabled` attribute will auto-enable indexing for your
-convenience, even though the default value of the `enabled` attribute is defined as `"false"` in the XSD schema.
-This behaviour exists only on schemas starting with version 11.
+Example with `local-heap`:
 
 [source,xml,options="nowrap",subs=attributes+]
 ----
-include::config_examples/indexing.xml[]
+include::config_examples/indexing_near_real_time.xml[]
 ----
 
-Programmatic:
+Example with `filesystem`:
 
-[source,java]
+[source,xml,options="nowrap",subs=attributes+]
 ----
-import org.infinispan.configuration.cache.*;
-
-ConfigurationBuilder cacheCfg = ...
-cacheCfg.indexing().enable()
-      .addProperty("property name", "propery value")
+include::config_examples/indexing_near_real_time_fs.xml[]
 ----
 
-NOTE: When using programmatic configuration, indexing must be enabled explicitly by invoking the `enable()` method, as
-no auto-enabling mechanism exist for programmatic configuration as in the case of XML configuration.
-
-===== Index names
-
-Each property inside the `index` element is prefixed with the index name, for the index named `org.infinispan.sample.Car` the `directory_provider` is `local-heap`:
+[[query_massindexer]]
+=== Re-indexing
 
-[source,xml,options="nowrap",subs=attributes+]
-----
-include::config_examples/indexing_property.xml[]
-----
+Occasionally you might need to rebuild the indexs by reconstructing it from the data stored in the Cache. You need to rebuild the index if you change the definition of what is indexed on your types, or if you change for example some _Analyzer_ parameter, as Analyzers affect how the index is written. Also, you might need to rebuild the index if you had it destroyed by some system administration mistake. To rebuild the index just get a reference to the Indexer and start it; beware it might take some time as it needs to reprocess all data in the grid!
 
 [source,java]
 ----
-cacheCfg.indexing()
-   .enable()
-      .addProperty("org.infinispan.sample.Car.directory_provider", "local-heap")
-
+Indexer indexer = Search.getIndexer(cache);
+CompletionStage<Void> future = index.run();
 ----
 
-{brandname} creates an index for each entity existent in a cache, and it allows to configure those indexes independently.
-For a class annotated with `@Indexed`, the index name is the fully qualified class name, unless overridden with the
-`name` argument in the annotation.
+//TODO
+//[[query_indexless]]
+//==== Indexless
 
-In the snippet below, the default storage for all entities is `infinispan`, but `Boat` instances will be stored on `local-heap` in an index named
-`boatIndex`. `Airplane` entities will also be stored in `local-heap`. Any other entity's index will be configured with the property prefixed by `default`.
+//TODO
+//[[query_hybrid]]
+//==== Hybrid
 
-[source,java]
-----
-package org.infinispan.sample;
+== Querying
```

Comment:
I know this is developer content but would "Search" be a more common/simpler term than "Query"?

Perhaps "Searching the Data Store" or "Searching {brandname} Caches" might be more meaningful from a high level?

Reply:
I also like more "Searching" than "Querying", will change the references